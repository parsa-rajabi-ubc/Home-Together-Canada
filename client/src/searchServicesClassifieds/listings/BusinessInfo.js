/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.20
 *
 * @Description: Business Listing Container
 *
 */

import React from 'react';
import PropTypes from "prop-types";

const BUSINESS_INFO_TEXT = {
    TITLE: "Business Info",
    WEBSITE: "Website",
    REPORT: "Report Listing"
}

function BusinessInfo(props) {
    const {
        logo,
        businessName,
        address,
        website,
        phone,
        email
    } = props;


    return (
        <div className={"float-right"}>
            <img src={logo} alt={BUSINESS_INFO_TEXT.TITLE + "'s logo"} className={""}/>
            <div className={"text-center"}>
                <section className={"my-3"}>
                    <label className={"label-result text-lg"}>{BUSINESS_INFO_TEXT.TITLE} </label>
                    <p className={"my-2"}>{businessName}</p>
                </section>

                <section className={"my-3"}>
                    <p>{address.streetLine1}</p>
                    <p>{address.streetLine2}</p>
                    <p>{address.city}, {address.province} </p>
                    <p>{address.postalCode}</p>

                    <p className={"my-2"}>{phone}</p>
                    <a href={"mailto:" + email} target="_blank" rel="noopener noreferrer" className={"link"}>{email}</a>
                    <a href={"https://" + website} target="_blank" rel="noopener noreferrer"
                       className={"link"}>{BUSINESS_INFO_TEXT.WEBSITE}</a>
                </section>

                <button className={"btn btn-red my-10 text-base px-0 py-2"}>
                    {BUSINESS_INFO_TEXT.REPORT}
                </button>
            </div>


        </div>
    );
}

BusinessInfo.propTypes = {
    logo: PropTypes.string.isRequired,
    businessName: PropTypes.string.isRequired,
    address: PropTypes.shape({
        streetLine1: PropTypes.string.isRequired,
        streetLine2: PropTypes.string,
        city: PropTypes.string.isRequired,
        province: PropTypes.string.isRequired,
        postalCode: PropTypes.string.isRequired,
    }),
    website: PropTypes.string.isRequired,
    phone: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
}

export default BusinessInfo;
