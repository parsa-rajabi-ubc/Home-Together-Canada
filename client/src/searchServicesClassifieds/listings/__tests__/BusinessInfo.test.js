/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.20
 *
 * @Description: Business Info Test
 *
 */

import React from 'react';
import renderer from 'react-test-renderer'
import BusinessInfo from "../BusinessInfo";

describe('BusinessInfo', () => {
    describe('Container test', () => {
        it('should match snapshot test if business is not nationWide', () => {
            //given
            const props = {
                logo: "public/images/12.jpg",
                businessName: "Bob the builder",
                isNationWide: false,
                address: ({
                    streetLine1: "123 Kelowna",
                    streetLine2: "",
                    city: "Kelowna",
                    province: "BC",
                    postalCode: "X0X 0X0",
                }),
                website: "www.bob.com",
                phone: "123-413-1341",
                email: "Bob@Builder.ca"
            }
            //when
            const component = renderer.create(<BusinessInfo {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
        it('should match snapshot test if business is nationWide', () => {
            //given
            const props = {
                logo: "public/images/12.jpg",
                businessName: "Bob the builder",
                isNationWide: true,
                address: ({
                    streetLine1: null,
                    streetLine2: null,
                    city: null,
                    province: null,
                    postalCode: null,
                }),
                website: "www.bob.com",
                phone: "123-413-1341",
                email: "Bob@Builder.ca"
            }
            //when
            const component = renderer.create(<BusinessInfo {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });
});