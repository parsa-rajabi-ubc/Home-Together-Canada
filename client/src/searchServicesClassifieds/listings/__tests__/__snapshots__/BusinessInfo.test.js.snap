// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessInfo Container test should match snapshot test 1`] = `
<div
  className="relative text-right my-10 mx-32"
>
  <div
    className="inline-block text-left mx-10"
  >
    <section
      className="my-3"
    >
      <label
        className="label text-lg text-orange-600"
      >
        Business Info
         
      </label>
      <p
        className="my-2"
      >
        Bob the builder
      </p>
    </section>
    <section
      className="my-3"
    >
      <p>
        123 Kelowna
      </p>
      <p>
        
      </p>
      <p>
        Kelowna
        , 
        BC
         
      </p>
      <p>
        X0X 0X0
      </p>
      <p
        className="my-2"
      >
        123-413-1341
      </p>
      <a
        className="link"
        href="mailto:Bob@Builder.ca"
        rel="noopener noreferrer"
        target="_blank"
      >
        Bob@Builder.ca
      </a>
      <a
        className="link"
        href="https://www.bob.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Website
      </a>
    </section>
  </div>
  <img
    alt="Business Info's logo"
    className="inline-block align-top"
    src="public/images/12.jpg"
  />
</div>
`;
